File: 1790D.cpp
// D. Matryoshkas
// https://codeforces.com/problemset/problem/1790/D
// rating: 1200

#include <bits/stdc++.h>
using namespace std;

int main(){
    int t;
    cin >> t;

    while(t--){
        int n;
        cin >> n;

        vector<int> a(n);
        map<int, int> counter;

        for(int i = 0; i < n; ++i){
            cin >> a[i];
            counter[a[i]]++;
        }

        int toPrint = 0;
        for(auto& [num, freq] : counter){
            int prevFreq = counter.count(num - 1) ? counter[num - 1] : 0;
            toPrint += max(0, freq - prevFreq);
        }

        cout << toPrint << "\n";
    }
}

File: 1790D.py
# D. Matryoshkas
# https://codeforces.com/problemset/problem/1790/D
# rating: 1200

from collections import Counter

t = int(input())

for _ in range(t):
    n = int(input())
    a = list(map(int, input().split()))

    counter = Counter(a)

    toPrint = 0
    for num in sorted(counter):
        toPrint += max(0, counter[num] - counter.get(num - 1, 0))

    print(toPrint)

# tle